### ISSUE TITLE
Converting datetime with tz=datetime.timezone to EWSDateTime fails with TypeError

### ISSUE THREAD

**Issue by tommilligan** [ecederstrand/exchangelib#1018]
**Describe the bug**

Converting a python stdlib `datetime.datetime` instance to an `EWSDateTime`, where the source timezone is an instance of `datetime.timezone`, fails with a `TypeError`.

**To Reproduce**

```python
In [1]: from exchangelib import EWSDateTime

In [2]: from datetime import datetime, timezone

In [3]: date = datetime.now(tz=timezone.utc)

In [4]: EWSDateTime.from_datetime(date)
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
~/reinfer/env/lib/python3.8/site-packages/exchangelib/ewsdatetime.py in from_timezone(cls, tz)
    271         try:
--> 272             return {
    273                 cls.__module__.split('.')[0]: lambda z: z,

KeyError: 'datetime'

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-4-4f5f6826f0d5> in <module>
----> 1 EWSDateTime.from_datetime(date)

~/reinfer/env/lib/python3.8/site-packages/exchangelib/ewsdatetime.py in from_datetime(cls, d)
    120             tz = d.tzinfo
    121         else:
--> 122             tz = EWSTimeZone.from_timezone(d.tzinfo)
    123         return cls(d.year, d.month, d.day, d.hour, d.minute, d.second, d.microsecond, tzinfo=tz)
    124

~/reinfer/env/lib/python3.8/site-packages/exchangelib/ewsdatetime.py in from_timezone(cls, tz)
    279             }[tz_module](tz)
    280         except KeyError:
--> 281             raise TypeError('Unsupported tzinfo type: %r' % tz)
    282
    283     @classmethod

TypeError: Unsupported tzinfo type: datetime.timezone.utc
```

**Expected behavior**

As the source timezone is valid, I would expect it to correctly convert to EWSTimeZone, and thereby a valid EWSDateTime.

**Additional context**

```bash
$ python --version
Python 3.8.10
$ pip freeze | rg exchangelib
exchangelib==4.6.0
```

**Comment by tommilligan** [comment]
Potentially related to #1014?

I've submitted a PR which fixes this issue here: #1017

**Comment by ecederstrand** [comment]
Thanks for the contribution!
